bigqueue.go

```diff
@@ -0,0 +1,106 @@
+package bigqueue
+
+import (
+	"errors"
+	"fmt"
+	"path"
+)
+
+const (
+	cDataFileFmt  = "data_%d.dat"
+	cDataFileSize = 128 * 1024 * 1024
+	cInt64Size    = 8
+)
+
+var (
+	// ErrEmptyQueue is returned when peek/dequeue is performed on an empty queue
+	ErrEmptyQueue = errors.New("queue is empty")
+)
+
+// BigQueue implements IBigQueue interface using Go slices
+type BigQueue struct {
+	index     *QueueIndex
+	arenaList []*Arena
+	dir       string
+}
+
+// NewBigQueue constructs an instance of *BigQueue
+func NewBigQueue(dir string) (*BigQueue, error) {
+	index, err := NewQueueIndex(dir)
+	if err != nil {
+		return nil, err
```

Comment:
we should return a more informed error I think. `fmt.Errorf("error in creating index :: %v", err)`